src/storage-client/src/controller.rs

```diff
@@ -2149,7 +2156,10 @@ where
             //
             // Note that while collections are dropped, the `client` may already
             // be cleared out, before we do this post-processing!
-            if cluster_id.is_some() && frontier.is_empty() {
+            if cluster_id.is_some()
+                && frontier.is_empty()
+                && self.state.collections.get(&id).is_some()
```

Comment:
You noticed that this broke something without the additional condition?

Reply:
Yes because sinks are on clusters and we move their frontiers to `[]`; if it isn't a collection, it isn't a source and shouldn't be handled with the other source drops.